### DISCUSSION TITLE
All media files give error 500 "File not found" after environment change to "prod"

### DISCUSSION THREAD

**Discussion by PanickingLynx** [sulu/sulu#6840] [Q&A]
**Hello sulu community,**
I am currently trying to go to prod with a sulu project. But no media files render if the environment variable "`APP_ENV`" is set to "`prod`".
The site works perfectly fine when it is set to "`dev`".
Any ideas on why that is?

**What I've tried so far:**

- Clearing the cache in both "`dev`" and "`prod`" environments with **console** and **adminconsole.**
- Reuploading the contents of the `/var/media/` folder.
- Checking for Database and routing config integrity.

Hope this is enough info for you guys to go on :) 

Best regards,
**_Panic_**

**Comment by alexander-schranz** [comment]
Please have a look at your log files always if you have an error.

There are mostly 3 different Log files, you can check when working with Symfony.

 - Application Logs: `/path/to/your-project/var/log`
 - Webserver Logs:
     - Nginx: `/var/log/nginx/error.log` or `/var/log/nginx/your-project.error.log`
     - Apache: `/var/log/apache2/error.log` or `/var/log/apache2/your-project.error.log`
 - PHP FPM Log: `/var/log/php8.1-fpm.log`

In one of this log files you should see why you are getting a 500 error.

Also check the Response and `HTTP Status Code` of the failed requeste, what is it, does the response contain any information?

In most cases if it is relatd to uploaded files the error ist mosty a wrong settings for your servers `file permissions` for the `var` and `public/uploads` directories. Check the file permission documentation here: [Sulu File Permission Documentation](https://docs.sulu.io/en/latest/cookbook/web-server/nginx.html#file-permissions).

**Reply by PanickingLynx** [reply]
Thanks for the quick reply.
I checked further on the Error and it states the following: "HttpCache should not forward any response without any cache expiration time to the store." When I follow the error response produced by Symfony.

I don't really understand how this is correlating to me changing the "dev" environment to "prod".

**Reply by alexander-schranz** [reply]
That should normally not happening and sounds like misconfiguretion in the `fos_http_cache.yml`, `sulu_http_cache.yaml`,  `public/index.php` or `Kernel.php` in context of HttpCache. 

Check what your project is different to a normal Sulu Skeleton: https://github.com/sulu/skeleton.

**Reply by PanickingLynx** [reply]
So I looked at the files and everything seems integral.
But what I forgot to mention was that I use the sulu Docker image. Would that make a difference?

**Reply by alexander-schranz** [reply]
Setup a new empty `sulu/skeleton` via `composer create-project sulu/skeleton` and check if there under prod the same erorr happens with your docker image. Else try it without a custom image and just the docker compose for (database)  and a local PHP env provided with the skeleton.

**Reply by PanickingLynx** [reply]
After further testing:
The application works on the environments: `dev` and `stage`.
Just not on `prod`.
So it is probably one of the `when@prod` directives that is messing me up, I am just not too sure which one.

**Reply by alexander-schranz** [reply]
Normally there is no diff between `stage` and `prod` in a Basic Sulu Setup only different urls in the webspace. A `stage` should always represent the same configs as a `prod`, as you want to test as near as possible to a prod env.
